File: src/test/java/com/jiro4989/tkfm/model/RectangleModelTest.java
package com.jiro4989.tkfm.model;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.*;

public class RectangleModelTest {
  @Test
  public void testConstructor() {
    var rect = new Rectangle(20, 40);
    assertEquals(20, rect.getWidth());
    assertEquals(40, rect.getHeight());
  }

  @Test
  public void testProperty() {
    var rect = new Rectangle(20, 40);
    assertEquals(20, rect.widthProperty().get());
    assertEquals(40, rect.heightProperty().get());

    rect.setWidth(100);
    rect.setHeight(200);
    assertEquals(100, rect.widthProperty().get());
    assertEquals(200, rect.heightProperty().get());
  }
}

File: src/test/kotlin/com/jiro4989/tkfm/model/RectangleModelTest.kt
package com.jiro4989.tkfm.model

import org.junit.jupiter.api.*
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.extension.ExtendWith
import org.testfx.framework.junit5.*

@ExtendWith(ApplicationExtension::class)
class RectangleModelTest {
  @Test
  fun testConstructor() {
    val rect = RectangleModel(20.0, 40.0)
    assertEquals(20.0, rect.width)
    assertEquals(40.0, rect.height)
  }

  @Test
  fun testProperty() {
    val rect = RectangleModel(20.0, 40.0)
    assertEquals(20.0, rect.width)
    assertEquals(40.0, rect.height)

    rect.width = 100.0
    rect.height = 200.0
    assertEquals(100.0, rect.width)
    assertEquals(200.0, rect.height)
  }
}
